src/rgw/rgw_user.cc

```diff
@@ -95,6 +95,47 @@ int rgw_user_sync_all_stats(RGWRados *store, const rgw_user& user_id)
 }
 
 /**
+ * Add group name to user info
+ */
+bool rgw_add_group_to_user(RGWUserInfo& info, string& group_name)
+{
+  if (std::find(info.groups.begin(), info.groups.end(), group_name)
+          == info.groups.end()) {
+    info.groups.push_back(std::move(group_name));
+    return true;
+  }
+  return false;
+}
+
+/**
+ * Remove group name from user info
+ */
+bool rgw_remove_group_from_user(RGWUserInfo& info, string& group_name)
+{
+  const auto& it = std::find(info.groups.begin(), info.groups.end(), group_name);
+  if (it != info.groups.end()) {
+    info.groups.erase(it);
+    return true;
+  }
+  return false;
+}
+
+/**
+ * Replace group name with new name in user info
+ */
+bool rgw_update_group_in_user(RGWUserInfo& info,
+                              string& group_name,
+                              string& new_group_name)
```

Comment:
`const string& group_name` since it's an input parameter. `new_group_name` should probably also be a const ref, and copied into `info.groups` instead of moved